components/meter/meter-linear.js

```diff
@@ -0,0 +1,53 @@
+import { html, LitElement } from 'lit-element/lit-element.js';
+import { meterStyles } from './meter-styles.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+class MeterLinear extends RtlMixin(LitElement)  {
+	static get styles() {
+		return [ meterStyles ];
+	}
+
+	render() {
+		return html `
+			<svg viewBox="-5 -4.5 ${this.width} 9" width="${this.width}">
+				<path class="d2l-meter-full-bar" d="M0,0 h ${this._actualWidth}" />
+				<path class="d2l-meter-progress-bar" d="M0,0 h ${this._computeProgressBarWidth(this.units, this.outOf, this._actualWidth)}" visibility="${this.units ? 'visable' : 'hidden'}"/>
+			</svg>
+		`;
+	}
+
+	static get properties() {
+		return {
+			units: { type: Number, reflect: true },
+			outOf: { type: Number, reflect: true, attribute: 'out-of' },
+			primary: { type: String, reflect: true },
+			secondary: { type: String, reflect: true },
+			width: { type: Number },
```

Comment:
What do we think about making the actual meter a block or inline-block element such that it would just take up the width of its container? That way you could go `<d2l-meter-linear style="width:100%">` or `<d2l-meter-linear style="width:200px">` to get whatever width you want? I imagine having it in a grid where it needs to fill its parent cell will be a common use case.

Reply:
I switched to the div method for now. I do plan on learning way more about svgs. Using Math to code is very awesome.

Reply:
I am also an svg newb but recently learned a bit to fix a couple of ragged icons.  One possibility is to split the bar into three paths - the ends and the middle.  This may allow growing of the middle section without skewing the rounded ends.  I think there might also be a couple of attributes that control how ends are treated on strokes, so there could be something there too.